### PR TITLE
Correctly inject subobjects parameter in logsdb tests

### PR DESCRIPTION
**Pull request by lkts**
Previously we would only set `subobjects` at a top level and then keep it the same in all objects. This PR adds variability in child objects. F.e. it is now possible to have a child object that introduces `subobjects: auto` or a child objects with `subobjects: true` when parent has `subobjects: auto`. 

There is one problem in this implementation - currently both standard and logsdb mapping is exactly the same. We probably want to also compare logsdb against "pure" standard mode (subobjects is default everywhere). Data generator does not really allow to do that now.

With these updates, there is a new test failure related to autoflattening so it needs to be merged after the fix.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/logsdb/datageneration/datasource/DefaultMappingParametersHandler.java [hunk 1]
```diff
@@ -90,12 +94,31 @@ public DataSourceResponse.ObjectMappingParametersGenerator handle(DataSourceRequ
...
+
             if (ESTestCase.randomBoolean()) {
                 parameters.put("dynamic", ESTestCase.randomFrom("true", "false", "strict", "runtime"));
             }
             if (ESTestCase.randomBoolean()) {
                 parameters.put("enabled", ESTestCase.randomFrom("true", "false"));
             }
+            // Changing subobjects from subobjects: false is not supported, but we can f.e. go from "auto" to "true".
+            if (ESTestCase.randomBoolean()) {
+                parameters.put("subobjects", ESTestCase.randomFrom(ObjectMapper.Subobjects.values()).toString());
```
Comment (martijnvg) [inline test/framework/src/main/java/org/elasticsearch/logsdb/datageneration/datasource/DefaultMappingParametersHandler.java:120]: Looks like this needs to randomization needs to be more restrictive? Given the following  test failure:

```
LogsDbVersusReindexedLogsDbChallengeRestIT > testMatchAllQuery FAILED
    org.elasticsearch.client.ResponseException: method [PUT], host [http://[::1]:61001], URI [/_index_template/baseline-template], status line [HTTP/1.1 400 Bad Request]
    {"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"composable template [baseline-template] template after composition is invalid"}],"type":"illegal_argument_exception","reason":"composable template [baseline-template] template after composition is invalid","caused_by":{"type":"illegal_argument_exception","reason":"invalid composite mappings for [baseline-template]","caused_by":{"type":"mapper_parsing_exception","reason":"Failed to parse mapping: Tried to add nested object [CIeekrBi] to object [Cj] which does not support subobjects","caused_by":{"type":"mapper_parsing_exception","reason":"Tried to add nested object [CIeekrBi] to object [Cj] which does not support subobjects"}}}},"status":400}
        at app//org.elasticsearch.client.RestClient.convertResponse(RestClient.java:351)
        at app//org.elasticsearch.client.RestClient.performRequest(RestClient.java:317)
        at app//org.elasticsearch.client.RestClient.performRequest(RestClient.java:292)
        at app//org.elasticsearch.datastreams.logsdb.qa.AbstractChallengeRestTest.createTemplates(AbstractChallengeRestTest.java:163)
        at app//org.elasticsearch.datastreams.logsdb.qa.AbstractChallengeRestTest.createTemplates(AbstractChallengeRestTest.java:110)
        at app//org.elasticsearch.datastreams.logsdb.qa.AbstractChallengeRestTest.beforeTest(AbstractChallengeRestTest.java:88)
        at java.base@22.0.1/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base@22.0.1/java.lang.reflect.Method.invoke(Method.java:580)
        at app//com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
        at app//com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:980)
        at app//com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
        at app//com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at app//org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at app//org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
        at app//org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at app//org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
        at app//org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at app//org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at app//org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at app//com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at app//com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
        at app//com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
        at app//com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
        at app//com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
        at app//com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
        at app//com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
        at app//com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
        at app//org.elasticsearch.test.cluster.local.DefaultLocalElasticsearchCluster$1.evaluate(DefaultLocalElasticsearchCluster.java:48)
        at app//org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at app//com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at app//org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
        at app//com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at app//com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at app//com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at app//com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at app//org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
        at app//org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at app//org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at app//org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at app//org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
        at app//org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at app//com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at app//com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
        at app//com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
        at java.base@22.0.1/java.lang.Thread.run(Thread.java:1570)
```

Comment (lkts) [reply test/framework/src/main/java/org/elasticsearch/logsdb/datageneration/datasource/DefaultMappingParametersHandler.java:120]: That's interesting.

Comment (lkts) [reply test/framework/src/main/java/org/elasticsearch/logsdb/datageneration/datasource/DefaultMappingParametersHandler.java:117]: I think i got this fixed, will re-run on top of latest fixes later.

Comment (kkrik-es) [reply test/framework/src/main/java/org/elasticsearch/logsdb/datageneration/datasource/DefaultMappingParametersHandler.java:117]: Stick to just overriding to "auto" for 20%?

Comment (lkts) [reply test/framework/src/main/java/org/elasticsearch/logsdb/datageneration/datasource/DefaultMappingParametersHandler.java:117]: But that does not work if parent is `false`, right?

**Comment by lkts** [comment]
So that's actually challenging because we use the same data generator instance. Previously it was never the case that the mapping should diverge between baseline and contender. I'll work on this but i think in the meantime we can still merge this.

File: test/framework/src/main/java/org/elasticsearch/logsdb/datageneration/datasource/DefaultMappingParametersHandler.java (hunk 1 above)
Comment (kkrik-es) [reply test/framework/src/main/java/org/elasticsearch/logsdb/datageneration/datasource/DefaultMappingParametersHandler.java:117]: Indeed..

**Review by kkrik-es — APPROVED** [review]

**Pull request merged**